src/core/basics.ml

```diff
@@ -205,3 +212,77 @@ let term_of_rhs : rule -> term = fun r ->
     TE_Some(Bindlib.unbox (Bindlib.bind_mvar vars p))
   in
   Bindlib.msubst r.rhs (Array.mapi fn r.vars)
+
+(** Total order on alpha-equivalence classes of terms not containing [Patt],
+   [TEnv] or [TRef].
+@raise Invalid_argument otherwise. *)
+let cmp_term : term Lplib.Extra.cmp =
```

Comment:
You didn't merge all of master yet, did you? Let me know when you will have merged all the master please. Thank you.

Reply:
Ah yes I have, mistakes slipped through

Reply:
It seems to me that the comparison rendered by github is weird

Reply:
What do you mean? You do not get the same thing on your machine?

Reply:
I obtain this diff using `git diff master -- src/core/*`
https://copycat.drycat.fr/?a5d32c929080ba32#D2SNs2W78NjEZGtKaXc1HuMHQLXCbTXWDxfNBBc3jJGH

I'm not sure why though

Reply:
Ah I understand. As the merges were not obvious, I used `git checkout --patch master <filepath>`, which directly applies patches to files to go incrementally (rather than merging everything at once). The result is that github is lost: the original version in the presented diff seem so concern master branch at the time I forked, not the new one.

My advice would be to use Emacs' ediff, or go with a plain diff file. It's a bit different, but it works quite well. On the other hand I don't know if I'm being a git heretic using that `checkout --patch` or if github is just not as handy as one would hope.

Reply:
I don't know. I never used --patch; I always use merge. What if you do merge origin/master now?

Reply:
One way may be to create a new pull request... I'll try out later on

Reply:
> I don't know. I never used --patch; I always use merge. What if you do merge origin/master now?

And I don't want to do that now because I'll have the 100+ test files to edit...

Reply:
If it's about porting the changes introduced by Rehan, a lot can be done automatically:
proof -> := begin
qed -> end
theorem -> opaque symbol
definition -> symbol

Reply:
Remark: translation to the new parser syntax should be mostly automatic too. Except for proof/begin, you can replace any command <c> by ;<c>.

Reply:
It should, but isn't: for instance, rules shouldn't have a terminator if there are other rules with `with`.

Reply:
I don't understand. What's the problem is you substitute "rule" by ";rule" (except on the first command of a file) and don't change "with"?

Reply:
True, I'll have to try.